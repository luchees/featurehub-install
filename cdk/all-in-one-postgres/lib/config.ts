import { InstanceClass, InstanceSize } from "@aws-cdk/aws-ec2";
import { PostgresEngineVersion } from "@aws-cdk/aws-rds";

interface FeaturehubConfig {
  [k: string]: {
    db: {
      dbName: string;
      userName: string;
      preferredMaintenanceWindow: string;
      version: PostgresEngineVersion;
      instanceClass: InstanceClass;
      instanceSize: InstanceSize;
    };
    fargate: {
      imageVersion: string;
      desiredCount: number;
      maxCapacity: number;
      minCapacity: number;
      cpu: number;
      memoryLimitMiB: number;
    };
  };
}

export const CONFIG: FeaturehubConfig = {
  dev: {
    db: {
      dbName: "featurehub",
      userName: "featurehub",
      preferredMaintenanceWindow: "Wed:10:00-Wed:11:00",
      version: PostgresEngineVersion.VER_13_3,
      instanceClass: InstanceClass.T3,
      instanceSize: InstanceSize.MEDIUM,
    },
    fargate: {
      imageVersion: "1.4.1",
      desiredCount: 2,
      maxCapacity: 8,
      minCapacity: 2,
      cpu: 256,
      memoryLimitMiB: 1024,
    },
  },
  staging: {
    db: {
      dbName: "featurehub",
      userName: "featurehub",
      preferredMaintenanceWindow: "Wed:10:00-Wed:11:00",
      version: PostgresEngineVersion.VER_13_3,
      instanceClass: InstanceClass.T2,
      instanceSize: InstanceSize.MICRO,
    },
    fargate: {
      imageVersion: "1.4.1",
      desiredCount: 1,
      maxCapacity: 2,
      minCapacity: 1,
      cpu: 256,
      memoryLimitMiB: 512,
    },
  },
  production: {
    db: {
      dbName: "featurehub",
      userName: "featurehub",
      preferredMaintenanceWindow: "Wed:10:00-Wed:11:00",
      version: PostgresEngineVersion.VER_13_3,
      instanceClass: InstanceClass.T3,
      instanceSize: InstanceSize.LARGE,
    },
    fargate: {
      imageVersion: "1.4.1",
      desiredCount: 2,
      maxCapacity: 8,
      minCapacity: 2,
      cpu: 256,
      memoryLimitMiB: 1024,
    },
  },
};
