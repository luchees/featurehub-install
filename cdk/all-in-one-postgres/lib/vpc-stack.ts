import * as cdk from "@aws-cdk/core";
import * as ec2 from "@aws-cdk/aws-ec2";

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  prefix?: string;
}

export class VpcStack extends cdk.Stack {
  readonly vpc: ec2.Vpc;

  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;

    this.vpc = new ec2.Vpc(this, `${prefix}-VPC`);
  }
}
