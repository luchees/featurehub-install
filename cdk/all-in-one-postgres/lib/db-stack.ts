import * as cdk from "@aws-cdk/core";
import * as ec2 from "@aws-cdk/aws-ec2";
import * as rds from "@aws-cdk/aws-rds";
import { CONFIG } from "./config";

/**
 * Database Stack for deploying Postgres DB for featurehub
 */

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  vpc: ec2.IVpc;
  dbCredentials: rds.Credentials;
  prefix?: string;
}

export class FeatureHubDatabaseStack extends cdk.Stack {
  readonly database: rds.DatabaseInstance;

  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;

    const config = CONFIG[props.stage];

    const database = new rds.DatabaseInstance(this, "FeaturehubDatabase", {
      databaseName: config.db.dbName,
      instanceIdentifier: `${prefix}-featurehub-db`,
      engine: rds.DatabaseInstanceEngine.postgres({
        version: config.db.version,
      }),
      removalPolicy: cdk.RemovalPolicy.RETAIN,
      credentials: props.dbCredentials,
      instanceType: ec2.InstanceType.of(
        config.db.instanceClass,
        config.db.instanceSize
      ),
      vpc: props.vpc,
      vpcSubnets: {
        subnets: props.vpc.isolatedSubnets,
      },
    });
    database.connections.allowFrom(
      ec2.Peer.ipv4(props.vpc.vpcCidrBlock),
      new ec2.Port({
        protocol: ec2.Protocol.TCP,
        stringRepresentation: "Postgres Port",
        fromPort: 5432,
        toPort: 5432,
      }),
      "Allow Access to featureHub database port"
    );

    this.database = database;
  }
}
