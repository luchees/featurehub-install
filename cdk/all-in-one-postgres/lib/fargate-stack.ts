import * as cdk from "@aws-cdk/core";
import * as ec2 from "@aws-cdk/aws-ec2";
import * as ecs from "@aws-cdk/aws-ecs";
import * as ecs_patterns from "@aws-cdk/aws-ecs-patterns";
import * as iam from "@aws-cdk/aws-iam";
import * as secretsmanager from "@aws-cdk/aws-secretsmanager";
import * as logs from "@aws-cdk/aws-logs";
import * as elb from "@aws-cdk/aws-elasticloadbalancingv2";
import { CONFIG } from "./config";

/**
 * Fargate Stack for deploying featurehub party-server
 */

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  prefix?: string;
  vpc: ec2.IVpc;
  dbSecret: secretsmanager.Secret;
  dbEndpoint: string;
  dbPort: string;
}

export class FeaturehubFargateStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const config = CONFIG[props.stage];

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;

    const cluster = new ecs.Cluster(this, "featurehubEcsCluster", {
      vpc: props.vpc,
      clusterName: `${prefix}-featurehub-cluster`,
      containerInsights: true,
    });

    const executionRolePolicy = new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      resources: ["*"],
      actions: ["logs:CreateLogStream", "logs:PutLogEvents"],
    });

    const taskDef = new ecs.FargateTaskDefinition(this, "featurehubTaskDef", {
      family: `${prefix}-featurehub-taskdef`,
    });
    taskDef.addToExecutionRolePolicy(executionRolePolicy);
    const loadBalancer = new elb.ApplicationLoadBalancer(
      this,
      "featurehubLoadBalancer",
      {
        vpc: props.vpc,
        loadBalancerName: `${prefix}-featurehub-lb`,
        internetFacing: true,
      }
    );

    const container = taskDef.addContainer(`${prefix}-featurehub`, {
      image: ecs.ContainerImage.fromRegistry(
        `featurehub/party-server:${config.fargate.imageVersion}`
      ),
      logging: new ecs.AwsLogDriver({
        logGroup: new logs.LogGroup(this, "featurehubContainerLogGroup", {
          logGroupName: `/ecs/${prefix}-featurehub`,
        }),
        streamPrefix: `${prefix}-featurehub`,
      }),
      environment: {
        UID: "999",
        GID: "999",
        "DB.URL": `jdbc:postgresql://${props.dbEndpoint}:${props.dbPort}/${config.db.dbName}`,
        "DB.USERNAME": config.db.userName,
        "JERSEY.CORS.HEADERS":
          "X-Requested-With,Authorization,Content-type,Accept-Version,Content-MD5,CSRF-Token,x-ijt,cache-control,x-featurehub,Baggage",
        "REGISTER.URL": `http://${loadBalancer.loadBalancerDnsName}/register-url?token=%s`,
      },
      secrets: {
        "DB.PASSWORD": ecs.Secret.fromSecretsManager(props.dbSecret),
      },
    });

    container.addPortMappings({
      containerPort: 8085,
      protocol: ecs.Protocol.TCP,
    });

    const fargateService = new ecs_patterns.ApplicationLoadBalancedFargateService(
      this,
      "featurehubFargate",
      {
        serviceName: `${prefix}-featurehub-fargate`,
        cluster: cluster,
        platformVersion: ecs.FargatePlatformVersion.VERSION1_4,
        taskDefinition: taskDef,
        healthCheckGracePeriod: cdk.Duration.minutes(5),
        desiredCount: config.fargate.desiredCount,
        loadBalancer,
        cpu: config.fargate.cpu,
        memoryLimitMiB: config.fargate.memoryLimitMiB,
      }
    );

    fargateService.targetGroup.configureHealthCheck({
      enabled: true,
      path: "/metric/health/readyness",
      port: "8085",
      healthyHttpCodes: "200",
    });

    //Cookie stickiness needs to be enabled to make sure caching works
    fargateService.targetGroup.enableCookieStickiness(cdk.Duration.days(1));

    const scaling = fargateService.service.autoScaleTaskCount({
      maxCapacity: config.fargate.maxCapacity,
      minCapacity: config.fargate.minCapacity,
    });
    scaling.scaleOnCpuUtilization("featurehubCpuScaling", {
      policyName: `${prefix}-featurehub-cpu-scaling`,
      targetUtilizationPercent: 80,
      scaleInCooldown: cdk.Duration.seconds(60),
      scaleOutCooldown: cdk.Duration.seconds(60),
    });
    scaling.scaleOnMemoryUtilization("featurehubMemScaling", {
      policyName: `${prefix}-featurehub-mem-scaling`,
      targetUtilizationPercent: 80,
      scaleInCooldown: cdk.Duration.seconds(60),
      scaleOutCooldown: cdk.Duration.seconds(60),
    });
  }
}
