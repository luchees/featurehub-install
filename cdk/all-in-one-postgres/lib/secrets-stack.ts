import * as cdk from "@aws-cdk/core";
import * as rds from "@aws-cdk/aws-rds";
import * as secretsmanager from "@aws-cdk/aws-secretsmanager";
import { CONFIG } from "./config";

/**
 * Secrets Stack for creating the Secrets to be used with Featurehub Database
 */

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  prefix?: string;
}

export class FeatureHubSecretsStack extends cdk.Stack {
  readonly dbPasswordSecret: secretsmanager.Secret;
  readonly dbCredentials: rds.Credentials;

  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;
    const config = CONFIG[props.stage];

    const dbPasswordSecret = new secretsmanager.Secret(
      this,
      "FeaturehubDbPassword",
      {
        secretName: `${prefix}-featurehub-db-password`,
        generateSecretString: {
          excludeCharacters: "%+~`#$&*()|[]{}:;<>?!/@ \\\"'", //password characters not supported by postgresql
        },
      }
    );
    // RDS Credentials to be used by RDS
    const dbCredentials = rds.Credentials.fromPassword(
      config.db.userName,
      dbPasswordSecret.secretValue
    );

    this.dbPasswordSecret = dbPasswordSecret;
    this.dbCredentials = dbCredentials;
  }
}
