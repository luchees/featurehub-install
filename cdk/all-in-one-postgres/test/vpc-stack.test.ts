import { SynthUtils } from "@aws-cdk/assert";
import * as cdk from "@aws-cdk/core";
import { VpcStack } from "../lib/vpc-stack";

describe("Featurehub vpc Stack", () => {
  test("snapshot", () => {
    const app = new cdk.App();
    const stack = new VpcStack(app, "MyTestVpcStack", {
      stage: "dev",
    });
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});
