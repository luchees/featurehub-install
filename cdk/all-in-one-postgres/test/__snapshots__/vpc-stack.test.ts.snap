// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featurehub vpc Stack snapshot 1`] = `
Object {
  "Resources": Object {
    "devVPC1AA5A1CF": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "devVPCIGWE80F99AC": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "devVPCPrivateSubnet1DefaultRouteED6B2E94": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "devVPCPublicSubnet1NATGatewayAC901F3E",
        },
        "RouteTableId": Object {
          "Ref": "devVPCPrivateSubnet1RouteTable94C03E5A",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "devVPCPrivateSubnet1RouteTable94C03E5A": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "devVPCPrivateSubnet1RouteTableAssociationAC478763": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "devVPCPrivateSubnet1RouteTable94C03E5A",
        },
        "SubnetId": Object {
          "Ref": "devVPCPrivateSubnet1SubnetE87A13E1",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "devVPCPrivateSubnet1SubnetE87A13E1": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "devVPCPrivateSubnet2DefaultRoute2E1F9DE4": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "devVPCPublicSubnet2NATGateway667CA9BE",
        },
        "RouteTableId": Object {
          "Ref": "devVPCPrivateSubnet2RouteTable58846670",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "devVPCPrivateSubnet2RouteTable58846670": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "devVPCPrivateSubnet2RouteTableAssociationA10B1144": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "devVPCPrivateSubnet2RouteTable58846670",
        },
        "SubnetId": Object {
          "Ref": "devVPCPrivateSubnet2Subnet1CEB740F",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "devVPCPrivateSubnet2Subnet1CEB740F": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "devVPCPublicSubnet1DefaultRouteDCEC8BBC": Object {
      "DependsOn": Array [
        "devVPCVPCGW6D377A17",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "devVPCIGWE80F99AC",
        },
        "RouteTableId": Object {
          "Ref": "devVPCPublicSubnet1RouteTable1A216D76",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "devVPCPublicSubnet1EIP9AD09B43": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "devVPCPublicSubnet1NATGatewayAC901F3E": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "devVPCPublicSubnet1EIP9AD09B43",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "devVPCPublicSubnet1Subnet5E7031F3",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "devVPCPublicSubnet1RouteTable1A216D76": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "devVPCPublicSubnet1RouteTableAssociation91008A4E": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "devVPCPublicSubnet1RouteTable1A216D76",
        },
        "SubnetId": Object {
          "Ref": "devVPCPublicSubnet1Subnet5E7031F3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "devVPCPublicSubnet1Subnet5E7031F3": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "devVPCPublicSubnet2DefaultRouteF43440BE": Object {
      "DependsOn": Array [
        "devVPCVPCGW6D377A17",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "devVPCIGWE80F99AC",
        },
        "RouteTableId": Object {
          "Ref": "devVPCPublicSubnet2RouteTableFA06D055",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "devVPCPublicSubnet2EIP46AEB48F": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "devVPCPublicSubnet2NATGateway667CA9BE": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "devVPCPublicSubnet2EIP46AEB48F",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "devVPCPublicSubnet2Subnet88BE79F5",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "devVPCPublicSubnet2RouteTableAssociation5F050B60": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "devVPCPublicSubnet2RouteTableFA06D055",
        },
        "SubnetId": Object {
          "Ref": "devVPCPublicSubnet2Subnet88BE79F5",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "devVPCPublicSubnet2RouteTableFA06D055": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "devVPCPublicSubnet2Subnet88BE79F5": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestVpcStack/dev-VPC/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "devVPCVPCGW6D377A17": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "devVPCIGWE80F99AC",
        },
        "VpcId": Object {
          "Ref": "devVPC1AA5A1CF",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
}
`;
